tests/common/plugins/sanity_check/__init__.py

```diff
@@ -269,6 +272,20 @@ def sanity_check(localhost, duthosts, request, fanouthosts, nbrhosts, tbinfo):
             pt_assert(False, "!!!!!!!!!!!!!!!! Post-test sanity check failed: !!!!!!!!!!!!!!!!\n{}"\
                 .format(json.dumps(post_failed_results, indent=4, default=fallback_serializer)))
 
+        logger.info("Compare with golden config db")
+        for duthost in duthosts:
+            if duthost.stat(path=GOLDEN_CONFIG).get('stat', {}).get('exists', False):
+                running_config = duthost.get_running_config_facts()
+                logger.debug("Running config of dut {}".format(running_config))
+
+                result = duthost.command("cat {}".format(GOLDEN_CONFIG))
+                golden_config = json.loads(result['stdout'])
+                logger.debug("Golden config of dut {}".format(golden_config))
+
+                if running_config != golden_config:
+                    duthost.command("cp {} {}".format(GOLDEN_CONFIG, CONFIG_DB))
+                    config_reload(duthost)
```

Comment:
This is not the correct place to add post test check.
Please create a new check by following the same pattern of other checks.
* Add a check item in the `checks.py` script.
* Based on the check results to do adaptive recovery. If golden config compare failed, the corresponding recovery method should be config reload.